helios-services/src/main/java/com/spotify/helios/master/MasterService.java

```diff
@@ -308,13 +316,40 @@ private void logBanner() {
    * @return A zookeeper client.
    */
   private ZooKeeperClient setupZookeeperClient(final MasterConfig config) {
+    ACLProvider aclProvider = null;
+    List<AuthInfo> authorization = null;
+
+    if (config.isZooKeeperEnableAcls()) {
+      final String masterPassword = config.getZooKeeperMasterPassword();
+      final String agentDigest = config.getZooKeeperAgentDigest();
+
+      if (masterPassword == null) {
+        throw new HeliosRuntimeException("ZooKeeper ACLs enabled but master password not set");
+      }
+
+      if (agentDigest == null) {
+        throw new HeliosRuntimeException("ZooKeeper ACLs enabled but agent digest not set");
+      }
+
+      final String masterDigest = Base64.toBase64String(Hash.sha1digest(
```

Comment:
It's an annoying function -- it throws a [checked] `java.security.NoSuchAlgorithmException` exception that has to be dealt with. I can use it if you feel strongly about it, but if so I'll probably wrap it in a function that re-throws that exception as a runtime exception.